src/funcs.c

```diff
@@ -1875,5 +1910,6 @@ static StructInitInfo module = {
 
 StructInitInfo * InitInfoFuncs ( void )
 {
+    funcsStateOffset = RegisterModuleState(0, InitModuleState, 0);
```

Comment:
hu, shouldn't the first argument be `sizeof(funcsModuleState)` ?

Reply:
Perhaps a convenience wrapper for this would be useful?
```C
#define REGISTER_MODULE_STATE(module, constr, destr) \
  module ## StateOffset = RegisterModuleState(sizeof(module ## ModuleState), constr, destr)
```
and then replace the above line by just
```C
  REGISTER_MODULE_STATE(InitModuleState, 0);
```

Reply:
Of course that might be confusing for all those modules which actually don't need spaces to store extra state, and which therefore couldn't use `REGISTER_MODULE_STATE`... so perhaps scratch that idea.

Reply:
I do like the fact that this macro makes the call to `RegisterModuleState` safe in terms of size allocation, though.